### ISSUE TITLE
invalid strftime in aws_srp.py

### ISSUE THREAD

**Issue by stephenoneal** [capless/warrant#21]
https://github.com/capless/warrant/blob/master/warrant/aws_srp.py#L158

I think you want "%a %b %d %H:%M:%S UTC %Y" for the strftime %-d is invalid.

**Comment by bjinwright** [comment]
I believe you are correct.

**Comment by armicron** [comment]
It will not authenticate with %d, as it will have a leading zero.
@stephenoneal what python version and OS do you use?

http://stackoverflow.com/questions/904928/python-strftime-date-without-leading-0

**Comment by stephenoneal** [comment]
python 3.4, Windows

**Comment by stephenoneal** [comment]
oddly i changed it to datetime.datetime.utcnow().strftime("%a %b %d %H:%M:%S UTC %Y") locally and it's passing the PASSWORD_VERIFIER challenge

**Comment by armicron** [comment]
>oddly i changed it to datetime.datetime.utcnow().strftime("%a %b %d %H:%M:%S UTC %Y") locally and it's passing the PASSWORD_VERIFIER challenge

It will fail on May 1.

**Comment by bjinwright** [comment]
Oh yeah, that happened at the beginning of this month.

**Comment by stephenoneal** [comment]
my attention on this is not great right now.  need more coffee..

import platform
windows_os = any(platform.win32_ver())

        timestamp = test_timestamp or \
                    ( datetime.datetime.utcnow().strftime("%a %b %#d %H:%M:%S UTC %Y") if windows_os \
                    else datetime.datetime.utcnow().strftime("%a %b %-d %H:%M:%S UTC %Y") )

**Comment by bjinwright** [comment]
@armicron I think @stephenoneal is proposing a way to detect the OS and format the string accordingly.

**Comment by armicron** [comment]
I can't test it on windows.
```python
import sys


if sys.platform.startswith('win32'):
    timestamp = test_timestamp or datetime.datetime.utcnow().strftime("%a %b %#d %H:%M:%S UTC %Y")
else:
    timestamp = test_timestamp or datetime.datetime.utcnow().strftime("%a %b %-d %H:%M:%S UTC %Y")
```

**Comment by bjinwright** [comment]
Fixed in #28
